src/msw/dirdlg.cpp

```diff
@@ -309,39 +259,94 @@ int wxDirDialog::ShowSHBrowseForFolder(WXHWND owner)
 //
 // Returns wxID_OK on success, wxID_CANCEL if cancelled by user or wxID_NONE if
 // an error occurred and we should fall back onto the old dialog.
-#if wxUSE_IFILEDIALOG
+#if wxUSE_IFILEOPENDIALOG
+
+int wxDirDialog::ShowIFileOpenDialog(WXHWND owner)
+{
+    HRESULT hr = S_OK;
+    wxCOMPtr<IFileOpenDialog> fileDialog;
+
+    if ( !InitIFileOpenDialog(m_message, m_path, HasFlag(wxDD_MULTIPLE),
+                              HasFlag(wxDD_SHOW_HIDDEN), fileDialog) )
+    {
+        return wxID_NONE; // Failed to initialize the dialog
+    }
+
+    hr = fileDialog->Show(owner);
+    if ( FAILED(hr) )
+    {
+        if ( hr == HRESULT_FROM_WIN32(ERROR_CANCELLED) )
+        {
+            return wxID_CANCEL; // the user cancelled the dialog
+        }
+        else
+        {
+            wxLogApiError(wxS("IFileDialog::Show"), hr);
+        }
+    }
+    else
+    if ( GetPathsFromIFileOpenDialog(fileDialog, HasFlag(wxDD_MULTIPLE),
```

Comment:
```suggestion
    else if ( GetPathsFromIFileOpenDialog(fileDialog, HasFlag(wxDD_MULTIPLE),
```
Just a style nitpick.